### ISSUE TITLE
About Model Analysis

### ISSUE THREAD

**Issue by Qinulinuli** [BojarLab/glycowork#33]
Dear Glycowork
  I saw the documentation stating that we can use analyze_ml_model to analyze machine learning models. Is there a similar method for analyzing deep learning.

**Comment by Bribak** [comment]
Hello & thanks for reaching out!
Currently, we don't have that for our deep learning models because it'd be a different method (i.e., they don't have straight-up feature importances like the scikit-learn type ML models have)

Our go-to general-purpose strategy for such analyses in the case of deep learning would be SHAP (https://shap.readthedocs.io/en/latest/) but I'm not yet fully convinced that we will integrate this into glycowork any time soon, for two reasons:
1) Adds a rather large dependency to the package for something that can be achieved with very little glue code (basically, train model with glycowork and then go to SHAP with that model for interpretation)
2) Less stream-lined than for the ML models, in that different DL model types might/will require dedicated handling in such an analysis, which would require some experimentation

So bottom line is I'd recommend you to take a look at SHAP and similar methods after having trained or retrieved a model from glycowork, it's quite handy to use and they have many tutorials:-)
We might add something like as a recommendation to the code snippets in the glycowork docs for 1.1

**Comment by Qinulinuli** [comment]
Thank you. Looking forward to your progress！
